### PR TITLE
Update to Byte Buddy 1.12.14

### PR DESCRIPTION
**Pull request by cushon**
This release includes the fix for the issue with visibility checks for
types in the default package that caused
https://github.com/apache/beam/issues/22749.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by cushon** [comment]
R: @kileys

**Comment by kileys** [comment]
Do you mind pulling in the test that caught this bug? 
https://github.com/apache/beam/issues/22784

**Comment by kileys** [comment]
Fixes https://github.com/apache/beam/issues/22749?

**Comment by kileys** [comment]
Run Python PreCommit

**Comment by kileys** [comment]
Run SQL PreCommit

**Comment by cushon** [comment]
> Do you mind pulling in the test that caught this bug?

Done: I added a regression test that uses the default package, and verified it fails with bytebuddy 1.12.13 and passes with version 1.12.14.

**Comment by kileys** [comment]
It looks like spotless doesn't like the missing package declaration. You'll need to exclude the file from running with spotless, but otherwise LGTM

**Comment by cushon** [comment]
> You'll need to exclude the file from running with spotless

Thanks, done

**Comment by cushon** [comment]
It looks like spotless crashes on files without package declarations (https://github.com/diffplug/spotless/issues/270), I pushed another commit to work around that

File: sdks/java/build-tools/src/main/resources/beam/suppressions.xml [hunk 1]
```diff
@@ -96,4 +96,6 @@
   <suppress files="org.apache.beam.sdk.io.thrift.TestThriftUnion.java" checks="[a-zA-Z0-9]*"/>
   <suppress files="org.apache.beam.sdk.io.thrift.payloads.*" checks="[a-zA-Z0-9]*"/>
 
+  <!-- Test deliberately uses the default package. -->
+  <suppress checks="PackageDeclaration" files="org.apache.beam.sdk.transforms.reflect.DefaultPackageTest.java"/>
```
Comment (kileys) [inline sdks/java/build-tools/src/main/resources/beam/suppressions.xml:100]: Do we need this even with the exclude?

Comment (cushon) [reply sdks/java/build-tools/src/main/resources/beam/suppressions.xml]: > Do we need this even with the exclude?

I don't think so, thanks, removed

**Comment by kileys** [comment]
It looks like it is needed for checkstyle...

**Comment by cushon** [comment]
> It looks like it is needed for checkstyle...

I added it back

**Comment by kileys** [comment]
Thanks, I'll merge once the tests pass

**Comment by kileys** [comment]
Run Java PreCommit

**Comment by kileys** [comment]
Run Go PreCommit

**Comment by kileys** [comment]
Run Java_Kafka_IO_Direct PreCommit

**Comment by kileys** [comment]
Run Java PreCommit

**Pull request merged**